packages/ra-core/src/reducer/admin/references/oneToMany.ts

```diff
@@ -13,12 +15,31 @@ interface State {
 
 type ActionTypes =
     | CrudGetManyReferenceSuccessAction
+    | CrudDeleteSuccessAction
     | { type: 'OTHER_ACTION'; payload: any; meta?: any };
 
 const oneToManyReducer: Reducer<State> = (
     previousState = initialState,
     action: ActionTypes
 ) => {
+    if (action.meta && action.meta.fetch === DELETE && action.meta.optimistic) {
+        const relatedTo = Object.keys(previousState).filter(key =>
```

Comment:
move into a removeOrphanReference function. This way future reader will know at once what we do on an optimistic delete

Reply:
Not an orphan actually but I see what you mean

Reply:
Yep, naming: the hardest part in programming